code/modules/vehicles/speedbike.dm

```diff
@@ -36,6 +36,362 @@
 	icon_state = "speedbike_red"
 	overlay_state = "cover_red"
 
+
+/obj/vehicle/space/speedbike/atmos
+	name = "prototype atmos vehicle"
+	desc = "This vehicle possesses unparalled utility for atmospherics containment and control"
+	icon_state = "atmo_bike"
+	overlay_state = "cover_atmo"
+	var/obj/machinery/portable_atmospherics/scrubber/huge/internal_scubber = null
+	var/obj/item/weapon/extinguisher/vehicle/internal_extinguisher = null
+	var/obj/machinery/portable_atmospherics/canister/CAN = null
+	var/loaded = TRUE
+
+/obj/vehicle/space/speedbike/atmos/New()
+	. = ..()
+	internal_scubber = new /obj/machinery/portable_atmospherics/scrubber/huge(src)
+	internal_scubber.on = 1
+	internal_extinguisher = new /obj/item/weapon/extinguisher/vehicle(src)
+	CAN = new /obj/machinery/portable_atmospherics/canister/oxygen(src)
+
+/obj/machinery/portable_atmospherics/canister/proto/oxygen
+	icon_state = "proto"
+	gas_type = "o2"
+	filled = 1
+	release_pressure = ONE_ATMOSPHERE*2
+
+/obj/item/weapon/extinguisher/vehicle
+	name = "extinguisher nozzle"
+	desc = "A heavy duty nozzle attached to a massive reserve tank."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "atmos_nozzle"
+	w_class = WEIGHT_CLASS_BULKY
+	safety = 0
+	max_water = 1000
+	power = 10
+	precision = 1
+	cooling_power = 7
+	var/vehicle = null
+
+/obj/item/weapon/extinguisher/vehicle/dropped(mob/user)
+	..()
+	user << "<span class='notice'>The fire hose snaps back into the [src]!</span>"
+	playsound(get_turf(src),'sound/items/change_jaws.ogg', 75, 1)
+	loc = vehicle
```

Comment:
Couldn't figure out why this wasn't working so I had to do drop shenanigans, thank you
\